test/jest.setup-file.ts

```diff
@@ -23,16 +28,18 @@ async function mockSign(
         }
         key.algorithm.name = 'RSA-PSS';
         try {
-            return RSABSSA.SHA384.PSSZero.Deterministic().blindSign(key, data);
+            // await is needed here because if the promised is returned, the algorithmName could be restored before the key is used, causing an error
+            return await RSABSSA.SHA384.PSSZero.Deterministic().blindSign(key, data);
         } finally {
             key.algorithm.name = algorithmName;
         }
     }
-    return webcrypto.subtle.sign(algorithm, key, data);
-}
 
-if (typeof crypto === 'undefined') {
-    Object.assign(global, { crypto: webcrypto });
+    // webcrypto calls crypto, which is mocked. We need to restore the original implementation.
+    crypto.subtle.sign = parentSign;
+    const res = crypto.subtle.sign(algorithm, key, data);
```

Comment:
Does this have the desired behavior? sign can be overwritten before the promise is resolved.

Should you do something like this?
```
res.finally(() => {
    crypto.subtile.sign = mockSign;
});
```

Reply:
updated